### PR TITLE
Set default k8s version to 1.24 and enable CronJobTimeZone feature gate

### PR DESCRIPTION
**Pull request by fiunchinho**
### What this PR does / why we need it

Towards https://github.com/giantswarm/roadmap/issues/2237
Towards https://github.com/giantswarm/roadmap/issues/2214

I'm adding the feature gate to all components (kubelet, controller-manager, scheduler, api-server) because that's [what we did for `vintage`](https://github.com/giantswarm/k8scloudconfig/pull/1109/files).

### Checklist

- [X] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create
/test upgrade
/run cluster-test-suites

### CODE REVIEW

File: helm/cluster-aws/templates/_control_plane.tpl [hunk 1]
```diff
@@ -128,10 +128,12 @@ spec:
           cloud-provider: aws
           allocate-node-cidrs: "true"
           cluster-cidr: {{ .Values.connectivity.network.podCidr }}
+          feature-gates: TTLAfterFinished=true,CronJobTimeZone=true
```
Comment (fiunchinho) [inline helm/cluster-aws/templates/_control_plane.tpl:131]: for the controller-manager

File: helm/cluster-aws/templates/_control_plane.tpl [hunk 2]
```diff
@@ -128,10 +128,12 @@ spec:
...
       scheduler:
         extraArgs:
           authorization-always-allow-paths: "/healthz,/readyz,/livez,/metrics"
           bind-address: 0.0.0.0
+          feature-gates: TTLAfterFinished=true,CronJobTimeZone=true
```
Comment (fiunchinho) [inline helm/cluster-aws/templates/_control_plane.tpl:136]: for the scheduler

File: helm/cluster-aws/templates/_control_plane.tpl [hunk 3]
```diff
@@ -178,6 +181,7 @@ spec:
       nodeRegistration:
         kubeletExtraArgs:
           cloud-provider: aws
+          feature-gates: CronJobTimeZone=true
```
Comment (fiunchinho) [inline helm/cluster-aws/templates/_control_plane.tpl:184]: does anyone know if feature gates need to be specified in the `joinConfiguration` if they are already defined in `initConfiguration`?

File: helm/cluster-aws/values.yaml [hunk 4]
```diff
@@ -54,7 +54,7 @@ controlPlane:
   oidc: {}
   rootVolumeSizeGB: 120
 internal:
-  kubernetesVersion: 1.23.16
+  kubernetesVersion: 1.24.10
```
Comment (fiunchinho) [inline helm/cluster-aws/values.yaml:57]: the feature gate is only available on 1.24+

File: helm/cluster-aws/templates/_control_plane.tpl [hunk 5]
```diff
@@ -137,6 +136,9 @@ spec:
           extraArgs:
             listen-metrics-urls: "http://0.0.0.0:2381"
             quota-backend-bytes: "8589934592"
+      feature-gates:
```
Comment (fiunchinho) [inline helm/cluster-aws/templates/_control_plane.tpl:139]: Moving the feature gates to the shared section, the gates are passed to all components

```
root@ip-10-16-16-221:/home/giantswarm# cat /etc/kubernetes/manifests/kube-scheduler.yaml | grep feature
    - --feature-gates=TTLAfterFinished=true,CronJobTimeZone=true

root@ip-10-16-16-221:/home/giantswarm# cat /etc/kubernetes/manifests/kube-apiserver.yaml | grep feature
    - --feature-gates=TTLAfterFinished=true,CronJobTimeZone=true

root@ip-10-16-16-221:/home/giantswarm# cat /etc/kubernetes/manifests/kube-controller-manager.yaml | grep feature
    - --feature-gates=TTLAfterFinished=true,CronJobTimeZone=true
```

**Review by bdehri — APPROVED** [review]

**Pull request merged**